solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -496,17 +496,29 @@ contract RandomBeacon is Ownable {
 
     /// @notice Approves DKG result. Can be called after challenge period for the
     ///         submitted result is finished. Considers the submitted result as
-    ///         valid, pays reward to the result submitter and completes the
-    ///         group creation by activating the candidate group.
+    ///         valid, pays reward to the result submitter, punishes misbehaved
+    ///         group members and completes the group creation by activating the
+    ///         candidate group.
     function approveDkgResult() external {
         dkg.approveResult();
-        // Pay the DKG result submission reward.
         tToken.safeTransfer(dkg.resultSubmitter, dkgResultSubmissionReward);
+
+        uint32[] memory misbehavedMembers = groups.getLatestMisbehavedMembers();
+        if (misbehavedMembers.length > 0) {
+            //TODO: Replace 2 weeks with governable parameter
```

Comment:
Yes, I think it can be reused.
If #2707 is merged first, I will update this parameter.
If this PR is merged first, then we can merge it later (I suggest we keep the TODO for now).

Reply:
I just approved #2707, should get merged once all checks pass.

Reply:
Let's use the same value - `sortitionPoolRewardsBanDuration`.